File: backend/app/schemas/product.py
from pydantic import BaseModel
from typing import List, Optional
from datetime import datetime
from .base import BaseSchema
from .category import Category
from .product_detail import ProductDetail
from .market import Market

class ProductDetailBase(BaseModel):
    product_id: int
    market_id: int
    price: float
    expiration_date: Optional[datetime] = None
    calories: Optional[float] = None

class ProductDetailCreate(ProductDetailBase):
    pass

class ProductDetail(ProductDetailBase):
    id: int
    market: Optional[Market] = None

    class Config:
        from_attributes = True

class ProductPriceBase(BaseModel):
    price: float
    market_id: int
    product_id: int

class ProductPriceCreate(ProductPriceBase):
    pass

class ProductPrice(ProductPriceBase):
    id: int
    last_updated: datetime

    class Config:
        from_attributes = True

class ProductBase(BaseModel):
    name: str
    description: Optional[str] = None
    brand: Optional[str] = None
    barcode: Optional[str] = None
    image_url: Optional[str] = None

class ProductCreate(ProductBase):
    category_ids: Optional[List[int]] = None

class ProductUpdate(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None
    brand: Optional[str] = None
    barcode: Optional[str] = None
    category_ids: Optional[List[int]] = None

class ProductInDB(ProductBase, BaseSchema):
    pass

class Product(ProductInDB):
    id: int
    details: List[ProductDetail] = []
    category_ids: List[int] = []
    categories: List[Category] = []

    class Config:
        from_attributes = True 

File: frontend/src/types/product.ts
export interface Product {
  id: number;
  name: string;
  description: string;
  brand: string;
  barcode: string;
  image_url: string;
  created_at: string;
  updated_at: string;
  details: ProductDetail[];
  category: Category;
  is_favorite?: boolean;
}

export interface ProductDetail {
  id: number;
  product_id: number;
  market_id: number;
  price: number;
  unit: string;
  expiration_date: string;
  calories: number;
  is_favorite: boolean;
  created_at: string;
  updated_at: string;
  market: Market;
}

export interface Market {
  id: number;
  name: string;
  logo_url?: string;
  website?: string;
}

export interface Category {
  id: number;
  name: string;
  description: string;
  parent_id: number | null;
  created_at: string;
  updated_at: string;
} 